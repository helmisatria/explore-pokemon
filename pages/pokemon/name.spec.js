import { createLocalVue, shallowMount } from '@vue/test-utils'
import { PiniaVuePlugin } from 'pinia'
import { createTestingPinia } from '@pinia/testing'
import PokemonDetailPage from './:name.vue'

import pokemonDetailData from '~/fixtures/pokemon-detail.json'

describe('Page: Pokemon Detail Page', () => {
  let wrapper

  const localVue = createLocalVue()
  localVue.use(PiniaVuePlugin)

  const generateWrapper = () => {
    return shallowMount(PokemonDetailPage, {
      localVue,
      pinia: createTestingPinia(),
      mocks: {
        $router: { go: jest.fn() },
      },
    })
  }

  beforeEach(() => {
    wrapper = generateWrapper()
  })

  it('should mount component properly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should show pokemon species info, given data fetched', async () => {
    wrapper.vm.$pinia.state.value.pokemon.pokemonDetail = pokemonDetailData.species[0]
    await wrapper.vm.$nextTick()

    expect(wrapper.find('.pokemon-title').exists()).toBe(true)
    expect(wrapper.find('.pokemon-title').text()).toBe('charmander #004')
    expect(wrapper.find('main').exists()).toBe(true)
    expect(wrapper.find('main').element).toMatchSnapshot()
    expect(wrapper.find('PokemonErrorInfo-stub').exists()).toBe(false)
  })

  it('should display error state, given failed fetch pokemon detail', async () => {
    wrapper.vm.$pinia.state.value.pokemon.fetchFailed = 'pokemon-detail'
    await wrapper.vm.$nextTick()

    expect(wrapper.find('PokemonErrorInfo-stub').exists()).toBe(true)
    expect(wrapper.find('main').exists()).toBe(false)
  })
})
