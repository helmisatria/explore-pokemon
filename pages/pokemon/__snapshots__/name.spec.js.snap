// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: Pokemon Detail Page should mount component properly 1`] = `
<div
  class=""
>
  <header
    class="flex items-center shadow-sm sticky top-0 backdrop-blur-sm"
  >
    <router-link
      class="back-btn pl-6 pr-4 py-5"
      to="/"
    >
      <icoback-stub />
    </router-link>
     
    <!---->
  </header>
   
  <!---->
</div>
`;

exports[`Page: Pokemon Detail Page should show pokemon species info, given data fetched 1`] = `
<main
  class="px-6 pt-5 pb-10"
>
  <div
    class="flex justify-center"
  >
    <img
      alt="charmander"
      class="w-[300px] h-auto"
      height="300"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/4.png"
      width="300"
    />
  </div>
   
  <article>
    <p
      class="mt-6"
    >
      Obviously prefers
hot places. When
it rains, steamis said to spout
from the tip of
its tail.
    </p>
  </article>
   
  <div
    class="mt-6"
  >
    <h2
      class="font-bold"
    >
      Type
    </h2>
     
    <pokemonlabeltypes-stub
      class="mt-3"
      types="[object Object]"
      variation="normal"
    />
  </div>
   
  <div
    class="mt-6"
  >
    <h2
      class="font-bold"
    >
      Stats
    </h2>
     
    <pokemonstats-stub
      class="mt-3"
      stats="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
    />
  </div>
</main>
`;
